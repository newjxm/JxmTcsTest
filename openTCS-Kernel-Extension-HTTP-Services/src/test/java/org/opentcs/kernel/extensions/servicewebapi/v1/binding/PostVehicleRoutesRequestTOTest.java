/**
 * Copyright (c) The openTCS Authors.
 *
 * This program is free software and subject to the MIT license. (For details,
 * see the licensing information (LICENSE.txt) you should have received with
 * this copy of the software.)
 */
package org.opentcs.kernel.extensions.servicewebapi.v1.binding;

import java.util.List;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.opentcs.kernel.extensions.servicewebapi.JsonBinder;

/**
 * Unit tests for {@link PostVehicleRoutesRequestTO}.
 */
class PostVehicleRoutesRequestTOTest {

  private JsonBinder jsonBinder;

  @BeforeEach
  void setUp() {
    jsonBinder = new JsonBinder();
  }

  @Test
  void jsonSample() {
    Approvals.verify(
        jsonBinder.toJson(
            new PostVehicleRoutesRequestTO(List.of("C", "F"))
                .setSourcePoint("A")
                .setResourcesToAvoid(List.of("A", "B"))
        )
    );
  }
}
