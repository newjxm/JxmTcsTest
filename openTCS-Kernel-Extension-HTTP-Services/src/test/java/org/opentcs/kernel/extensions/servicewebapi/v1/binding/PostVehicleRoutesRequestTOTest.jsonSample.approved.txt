{
  "destinationPoints" : [ "C", "F" ],
  "sourcePoint" : "A",
  "resourcesToAvoid" : [ "A", "B" ]
}